packages/itwinui-react/src/core/Table/columns/actionColumn.tsx

```diff
@@ -94,45 +102,41 @@ export const ActionColumn = <T extends Record<string, unknown>>({
               });
             };
             return (
-              <MenuItem
+              <Checkbox
                 key={column.id}
-                startIcon={
-                  <Checkbox
-                    checked={checked}
-                    disabled={column.disableToggleVisibility}
-                    onClick={(e) => e.stopPropagation()}
-                    onChange={onClick}
-                    aria-labelledby={`iui-column-${column.id}`}
-                  />
-                }
-                onClick={onClick}
+                checked={checked}
                 disabled={column.disableToggleVisibility}
-              >
-                <div id={`iui-column-${column.id}`}>
-                  {column.render('Header')}
-                </div>
-              </MenuItem>
+                onClick={(e) => e.stopPropagation()}
+                onChange={onClick}
+                label={column.render('Header')}
+              />
             );
           });
 
-      const dropdownMenuProps =
+      const popoverProps =
         typeof columnManager !== 'boolean'
           ? columnManager.dropdownMenuProps
           : {};
 
       return (
-        <DropdownMenu
-          {...dropdownMenuProps}
-          menuItems={headerCheckBoxes}
+        <Popover
+          content={
+            <Surface as={'fieldset'}>
```

Comment:
1. There is no need for `Surface` here. You can use `<Popover applyBackground>`.
2. You are missing a `<legend>`. A fieldset without legend is useless.
3. We need to add some CSS to reset the default styles of `<fieldset>`.


Something like:

```jsx
<Popover
  applyBackground
  {...popoverProps}
  content={
    <fieldset>
      <VisuallyHidden as='legend'>Show/hide columns</VisuallyHidden>
      {headerCheckBoxes}
    </fieldset>
  }
>
```

Reply:
Re: 2, the `<legend>` needs to be a direct child of `<fieldset>` for it to be properly associated. It shouldn't be inside `Flex`.

Re: 3, I don't think we need to maintain this complex CSS for sizing it to be exactly 8.5 items, especially since we want to add more things in this popover in the future. I would say lets just pick an arbitrary max-height and remove all those CSS variables.